pkg/virtctl/memorydump/memorydump.go

```diff
@@ -0,0 +1,367 @@
+
+  #Create and download memory dump to the given output file.
+  {{ProgramName}} memory-dump get myvm --claim-name=memoryvolume --create-claim --output=memoryDump.dump.gz
+
+  #Dump memory again to the same virtual machine with an already associated pvc(existing memory dump on vm status).
+  {{ProgramName}} memory-dump get myvm
+
+  #Download the last memory dump associated on the vm 'myvm' to the given output file.
+  {{ProgramName}} memory-dump download myvm --output=memoryDump.dump.gz
+
+  #Remove the association of the memory dump pvc (to be able to dump to another pvc).
+  {{ProgramName}} memory-dump remove myvm
+  `
+	return usage
+}
+
+// NewMemoryDumpCommand returns a cobra.Command to handle the memory dump process
+func NewMemoryDumpCommand(clientConfig clientcmd.ClientConfig) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "memory-dump get/download/remove (VM)",
+		Short:   "Dump the memory of a running VM to a pvc",
+		Example: usageMemoryDump(),
+		Args:    templates.ExactArgs("memory-dump", 2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			c := command{clientConfig: clientConfig}
+			return c.run(args)
+		},
+	}
+	cmd.SetUsageTemplate(templates.UsageTemplate())
+	cmd.Flags().StringVar(&claimName, claimNameArg, "", "pvc name to contain the memory dump")
+	cmd.Flags().BoolVar(&createClaim, createClaimArg, false, "Create the pvc that will conatin the memory dump")
+	cmd.Flags().StringVar(&storageClass, storageClassArg, "", "The storage class for the PVC.")
+	cmd.Flags().StringVar(&accessMode, accessModeArg, "", "The access mode for the PVC.")
+	cmd.Flags().StringVar(&outputFile, "output", "", "Specifies the output path of the memory dump to be downloaded.")
+
+	return cmd
+}
+
+func (c *command) run(args []string) error {
+	namespace, _, err := c.clientConfig.Namespace()
+	if err != nil {
+		return err
+	}
+	virtClient, err := kubecli.GetKubevirtClientFromClientConfig(c.clientConfig)
+	if err != nil {
+		return fmt.Errorf("cannot obtain KubeVirt client: %v", err)
+	}
+
+	vmName := args[1]
+	switch args[0] {
+	case "get":
+		return getMemoryDump(namespace, vmName, virtClient)
+	case "download":
+		return downloadMemoryDump(namespace, vmName, virtClient)
+	case "remove":
+		return removeMemoryDump(namespace, vmName, virtClient)
+	default:
+		return fmt.Errorf("invalid action type %s", args[0])
+	}
+}
+
+func calcMemoryDumpExpectedSize(vmName, namespace string, virtClient kubecli.KubevirtClient) (*resource.Quantity, error) {
+	vmi, err := virtClient.VirtualMachineInstance(namespace).Get(vmName, &metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	return kutil.CalcExpectedMemoryDumpSize(vmi), nil
+}
+
+func calcPVCNeededSize(memoryDumpExpectedSize *resource.Quantity, storageClass *string, virtClient kubecli.KubevirtClient) (*resource.Quantity, error) {
+	cdiConfig, err := virtClient.CdiClient().CdiV1beta1().CDIConfigs().Get(context.Background(), configName, metav1.GetOptions{})
```

Comment:
Same as above. Maybe we should also check if CDI (not only the config) exists.

Reply:
Same answer, it will always be the hard coded name.